subsys/bluetooth/host/id.c

```diff
@@ -361,14 +380,21 @@ int bt_id_set_adv_private_addr(struct bt_le_ext_adv *adv)
 		return 0;
 	}
 
-	err = bt_rpa_create(bt_dev.irk[adv->id], &rpa);
-	if (!err) {
-		err = bt_id_set_adv_random_addr(adv, &rpa);
-		if (!err) {
-			atomic_set_bit(adv->flags, BT_ADV_RPA_VALID);
+	bt_le_ext_adv_foreach(adv_is_rpa_generated_for_adv_sets_with_same_id, &rpa_data);
+
+	if (!bt_addr_cmp(&rpa, BT_ADDR_NONE)) {
```

Comment:
Looks like we should add `bt_addr_eq()` just like we have `bt_addr_le_eq()`. @alwa-nordic any chance you could do that? Anyway, while it's missing it'd be more correct do do `bt_addr_cmp() == 0` since the function returns an int and not a bool (MISRA-C requires that conditional statements use strict boolean expressions rather than treating pointers or ints like bool variables)

Reply:
> Anyway, while it's missing it'd be more correct do do bt_addr_cmp() == 0 since the function returns an int and not a bool (MISRA-C requires that conditional statements use strict boolean expressions rather than treating pointers or ints like bool variables)

I generally agree with that, but since we have in Zephyr not yet transitioned to the stage where we should block PRs on that, it should be fine to still do `!`. AFAIK we are still on "Stage I" from https://docs.zephyrproject.org/latest/contribute/coding_guidelines/index.html#coding-guidelines

Reply:
PR for `bt_addr_eq`: https://github.com/zephyrproject-rtos/zephyr/pull/57658